test/workflow/Workflow.test.js

```diff
@@ -9,6 +9,25 @@ import { Fetcher } from '../../src/index.js';
 describe('Workflow', function () {
   this.timeout(30 * 1000);
 
+  it('should run the deep crawler correctly', async () => {
+    const wf = fox
+      .init('https://pokemondb.net/pokedex/national')
+      .deepcrawl('scrape pokemon with name, weight, and a list of traits')
+      .limit(4);
+
+    const results = await wf.run();
+    const urls = results.items.map((it) => it._url);
+
+    const expected = [
+      'https://pokemondb.net/pokedex/bulbasaur',
+      'https://pokemondb.net/pokedex/ivysaur',
+      'https://pokemondb.net/pokedex/venusaur',
+      'https://pokemondb.net/pokedex/charmander',
+    ];
+
+    assert.equal(JSON.stringify(urls), JSON.stringify(expected));
+  });
+
```

Comment:
can we also add a test below this that starts on the wrong page, and checks that it navigates to the right page?


```
  it('should deep crawl from wrong page @run @fast', async () => {
    const wf = fox
      .config({ cache: testCache() })
      .init('https://pokemondb.net/pokedex/national')
      .deepcrawl('find pokemon game locations and their names')
      .limit(4);

    const results = await wf.run();
    const urls = results.items.map((it) => it._url);

    const expected = [
//...eg....
'https://pokemondb.net/location',
'https://pokemondb.net/location#tab=loc-johto',
'https://pokemondb.net/location#tab=loc-hoenn',
'https://pokemondb.net/location#tab=loc-sinnoh',
    ];

    assert.equal(JSON.stringify(urls), JSON.stringify(expected));
  });
```